src/components/DatePicker/CalendarPicker/generateMonthMatrix.js

```diff
@@ -56,5 +55,11 @@ export default function generateMonthMatrix(year, month) {
         }
         matrix.push(currentWeek);
     }
+
+    // Add null values for days before the first day of the month
+    for (let i = matrix[0].length; i < 7; i++) {
+        matrix[0].unshift(null);
+    }
+
```

Comment:
I am on mobile so I missed on thing. The snippet I pasted above (old code) would go from 0 (start of the week assumed earlier - which is now wrong) until the first week day of the month (which before would always be 0 or higher). Now, because week starts on 1 (Sunday) the first week day of a month could be Sunday (0) and the loop would break and cause infinite loop. My change (unshift) accounts for any start of the week and any first week day of the month.